Update and simplify typechecking-faqs.md

robertknight opened the pull request:
Update typechecking-faqs.md to reflect the fact that we now author production code in TypeScript instead of JSDoc comments.

This obsoletes a number of the questions which were focused on dealing with the less-well documented JSDoc syntax.

The section at the end about `useRef` and `useState` types has become obsolete as a result of the switch to TS, which makes typing easier, and also changes in Preact which made the comment about how `useRef<T|null>(null)` works incorrect. The preferred way to use `useRef`/`useState` now is just to use them in the most obvious way like any other function. Hence no special guidance is needed.

robertknight committed:
Update and simplify typechecking-faqs.md

Update typechecking-faqs.md to reflect the fact that we now author production code in TypeScript instead of JSDoc comments.

These obsoletes a number of the questions which were focused on dealing with the less-well documented JSDoc syntax.

The section at the end about `useRef` and `useState` types has become obsolete as a result of the switch to TS, which makes typing easier, and also changes in Preact which made the comment about how `useRef<T|null>(null)` works incorrect. The preferred way to use `useRef`/`useState` now is just to use them in the most obvious way like any other function. Hence no special guidance is needed.

acelaya commented on docs/typechecking-faqs.md:
```diff
@@ -1,64 +1,39 @@
- * @param {MyButtonProps} props
- */
-function MyButton({ label }) {
+function MyButton({ label }: MyButtonProps) {
```
Perhaps the `MyButton` component and the `MyButtonProps` type should be exported in this example, to differentiate from the example below, which specifically mentions non-exported symbols.

```suggestion
export default function MyButton({ label }: MyButtonProps) {
```

acelaya commented on docs/typechecking-faqs.md:
````diff
@@ -1,64 +1,39 @@
-}
-```
+```tsx
+type MyButtonProps = {
+  /** Text label for the button. */
+  label: string;
+};
````
Perhaps the `MyButton` component and the `MyButtonProps` type should be exported in this example, to differentiate from the example below, which specifically mentions non-exported symbols.

```suggestion
export type MyButtonProps = {
  /** Text label for the button. */
  label: string;
};
```

robertknight committed:
Update typechecking-faqs.md

Use an exported component for the props example. Per PR feedback, use an exported component in the first example, as this is the most common use case.
